### PR TITLE
[PM-2320] Improve Android block Auto-fill URIs

### PR DESCRIPTION
**Pull request by fedemkr**
## Type of change
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective

Improve current way to block Auto-fill URIs. This is an Android feature.

## Code changes

* **BlockAutofillUrisPage.xaml:** New view to display blocked autofill URIs.
* * **BlockAutofillUrisPage.xaml.cs:** New view to display blocked autofill URIs. Added this as theme dirtable to change the placeholder corresponding to the current app theme.
* **BlockAutofillUrisPageViewModel.cs:** New view model to handle the logic for blocked autofill URIs and also validation on input.
* **OptionsPage.xaml:** Changed UI to be simpler and just navigate to the new view
* **OptionsPage.xaml.cs:** Removed old blocked Uris logic
* **OptionsPageViewModel.cs:** Removed old blocked Uris logic and added navigation to new view
* **DeviceActionService.cs:** Added new `DisplayValidatablePromptAsync(...)` method that shows a custom alert dialog with a custom view that has validations and execute the validations when pressing `save`. The dialog is dismissed on `save` only when all validations are valid. Also, added a few improvements for accessibility.
* **validatable_input_dialog_layout.xml:** Custom dialog view to display input that can be validated
* **dimens.xml:** Custom dimensions for `validatable_input_dialog_layout.xml`
* **CustomLabel.cs:** Added ability to change the font weight
* **CustomLabelRenderer.cs:** Added ability to change the font weight on API 28+
* **ValidatablePromptConfig.cs:** Config parameters for `DeviceActionService.DisplayValidatablePromptAsync`. Chose struct over method parameters to have more flexibility and because they are a lot.
* **empty_uris_placeholder.xml:** New icon to be displayed when there are no Uris in the list
* **empty_uris_placeholder_dark.xml:** New icon to be displayed when there are no Uris in the list on dark themes (transparent fill color)

### Options Block autofill light
Options Block autofill light

### Options Block autofill dark
Options Block autofill dark

### Block autofill empty light
Block autofill empty light

### Block autofill empty dark
Block autofill empty dark

### Block autofill multiple items light
Block autofill multiple items light

### Block autofill multiple items dark
Block autofill multiple items dark

### Block autofill new uri
Block autofill new uri

### Block autofill new uri error
Block autofill new uri error

### Block Autofill edit uri
Block Autofill edit uri

## Before you submit
- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

**Comment by danielleflinn** [comment]
@Federico-Maccaroni is there a way to embed the color variables from the theme file into the SVG? That way it will appear nicely in each theme. 
If so, `MutedColor` could be used for the darker outline color and the `ListHeaderBackgroundColor` for the lighter fill color.

**Review by LRNcardozoWDF — DISMISSED** [review]
Nice job @fedemkr, LGTM.

**Review by mpbw2 — APPROVED** [review]

File: src/App/Resources/AppResources.resx [hunk 1]
```diff
@@ -2643,4 +2640,50 @@ Do you want to switch to this account?</value>
...
+  <data name="BlockAutoFill" xml:space="preserve">
+    <value>Block auto-fill</value>
+  </data>
+  <data name="AutoFillWillNotBeOfferedForTheseURIs" xml:space="preserve">
+    <value>Auto-fill will not be offered for these URIs.</value>
+  </data>
+  <data name="NewBlockedURI" xml:space="preserve">
+    <value>New blocked URI</value>
+  </data>
+  <data name="URISaved" xml:space="preserve">
```
Comment (mpbw2) [inline src/App/Resources/AppResources.resx:2652]: @fedemkr Sorry I missed this, looks like `URISaved` made it in twice (see line 2674 below)

Comment (fedemkr) [reply src/App/Resources/AppResources.resx:2652]: Nice catch @mpbw2 , will have a PR ready right away

**Pull request merged**